### ISSUE TITLE
setVariant with Quantity not working

### ISSUE THREAD

**Issue by alaxicsmith** [Shopify/mobile-buy-sdk-ios#117]
Anyone having trouble getting the cart to add multiple items? I'm using Swift and the following line of code to have multiple items added to the cart: 

`self.cart.setVariant(product.variant, withTotalQuantity: 2)`

**Comment by dbart01** [comment]
Could you provide more details on the issues? What exactly is the problem with calling
`self.cart.setVariant(product.variant, withTotalQuantity: 2)` and what is the expectation?

Some things to consider:
+ does the product variant being added with quantity of 2 already exist in the cart?
+ what is the value of `self.cart.lineItems.count` and does it change as expected after adding a variant with quantity?
+ does the `self.cart.lineItems` array contain a `BUYCartLineItem` with a reference to `BUYProductVariant` that you are trying to add after calling `setVariant(_:withTotalQuantity:)`?

**Comment by alaxicsmith** [comment]
Yep, of course. 

The product being added to the cart with an increased quantity does already exist in the cart. 

The value of self.cart.lineItems before attempting to add another product to the cart is 1.  After attempting to add the same product to the cart with an increased quantity, the value is still 1. 

Yes.

**Comment by dbart01** [comment]
I suspect that there may be confusion in how `lineItems` work. Correct me if I'm wrong but it seems that you are expecting `self.cart.lineItems.count` to reflect the quantity of items? This is not the case.

The number of objects in the `lineItems` property does __not__ reflect the quantity of items. In fact, `quantity` is a property of `BUYLineItem`. The `lineItems` contain `BUYCartLineItem` objects which inherit from `BUYLineItem`. Please ensure that the quantity is accurately reflected in the `BUYCardLineItem` objects.

In other words, assuming you've only added one variant to the cart:
```swift
let lineItem = self.cart.lineItems[0]
if lineItem.quantity == 2 {
    // Should be true
} 
```

**Comment by alaxicsmith** [comment]
That works! Great, well now when I go to the checkout, it still only shows the quantity as one? How do I update the lineItems in the cart?

**Comment by dbart01** [comment]
You can use the same call from before to update the quantity of specific product variant:
```swift
self.cart.setVariant(product.variant, withTotalQuantity: 2)
```

**Comment by asudevil** [comment]
I'm having similar issues.  I'm following the example given in the integration guide/checkout.

`cart = client.modelManager.insertCartWithJSONDictionary(nil)`
//Button to add item
`addItemButtum.addTarget(self, action: #selector(self.addToCart(_:)), for controlEvents: . TouchUpInside)`
// My function to add item to cart
`func addToCart(button: UIButton)
  if let variant = product?.variants.firstObject as? BuyProductVariant {
    cart.addVariant(variant) 
}
print("Number of items: \(cart.lineItems.count)")`

The addToCart is overriding the previously added item instead of creating a new line.  How do I create a new lineItem

**Comment by bgulanowski** [comment]
@asudevil can you please open a new issue? We need to track each issue independently. Thanks.
